PSSM_score_Peptide/main.cpp

```diff
@@ -552,12 +552,14 @@ int assignPvalueToPSSMaRRAY(int argc, char *argv[])
 		if (place == -1) place = 0; //so that we get p value = 1 in this case.
 		//cout << "place = " << place << endl;
 		double p_Value = (double(numberOfRandomPSSM) - double(place)) / double(numberOfRandomPSSM);		
+		
+		double numberOfHitsNorm = numberOfHitsInRealPSSM;
 		if (useFactor & numberOfSeq != 0) {
-			float factor = float(1000000) / float(numberOfSeq);
-			numberOfHitsInRealPSSM *= factor;
+			double rpmFactorValue = double(1000000) / double(numberOfSeq);
+			numberOfHitsNorm *= rpmFactorValue;
 		}
 
-		listOfPvaluesFile << rpif._PSSM_array[i].PSSM_name << "\t" << p_Value << "\tTrue_Hits: " << numberOfHitsInRealPSSM <<endl; // << " total true hits " << numberOfHitsInRealPSSM << endl;
+		listOfPvaluesFile << rpif._PSSM_array[i].PSSM_name << "\t" << p_Value << "\tNorm_Hits: " << numberOfHitsNorm << "\tTrue_Hits: " << numberOfHitsInRealPSSM <<endl; // << " total true hits " << numberOfHitsInRealPSSM << endl;
```

Comment:
The code which merges the file extract hits by:
```cpp
hits = line_tokens[2].split()[-1]
```
This means that it will take the hits without normalization which isn't what we want if I understand correctly

Reply:
we take token in position 2, I put the Norm Hits before True Hits,
So it is take the right hit.